CMakeLists.txt

```diff
@@ -223,19 +237,36 @@ endif(ENABLE_MPI)
 
 #Generate CompiledDefines from libraries we found
 configure_file(scripts/compiledDefinesTemplate.hpp.in ${OCCA_BUILD_DIR}/include/occa/defines/compiledDefines.hpp)
+install(FILES ${OCCA_BUILD_DIR}/include/occa/defines/compiledDefines.hpp DESTINATION include/occa/defines)
```

Comment:
I may be showing my lack of cmake knowledge here, but what does this (and line 269 below) actually do? As far as I can tell they just set install directories to what they default to anyway.

Reply:
It tells CMake to actually install (copy) the referenced files/directories at install time. I think the library is something that is installed by default, but the (public) headers are not.
But looking at the recent CMake documentation it looks like there's a nicer way to do it, so I'll tweak this a bit.

Reply:
Hm, the nicer way I thought about does not really work. Looking at some other examples online something similar is done as above. But I'm very much open to suggestions about a better / more modern way of doing it.

Reply:
Sorry, I'm still not really seeing what this does. In particular, after building I see only include/occa/defines/compiedDefines.hpp in my build folder. Moreover, commenting out this line(and line 273 now) seems to make no difference in the contents of the build folder, or with the build itself. Are these lines even needed?

Reply:
nevermind, I think I figured it out. Messaged in slack.